geostore/tiles/mixins.py

```diff
@@ -120,41 +120,38 @@ def render_to_response(self, context, **response_kwargs):
         self.object = self.get_object()
         last_update = self.get_last_update()
 
-        if last_update:
-            cache_key = f'tilejson-{self.object.name}'
-            version = int(last_update.timestamp())
-            tilejson_data = cache.get(cache_key, version=version)
+        cache_key = f'tilejson-{self.object.name}'
+        version = int(last_update.timestamp())
+        tilejson_data = cache.get(cache_key, version=version)
 
-            if tilejson_data is None:
-                tilejson_data = json.dumps(self.get_tilejson())
-                cache.set(cache_key, tilejson_data, version=version)
+        if tilejson_data is None:
+            tilejson_data = json.dumps(self.get_tilejson())
+            cache.set(cache_key, tilejson_data, version=version)
 
-            response_kwargs.setdefault('content_type', self.content_type)
-            return self.response_class(
-                content=tilejson_data,
-                **response_kwargs,
-            )
-
-        return HttpResponseNotFound()
+        response_kwargs.setdefault('content_type', self.content_type)
+        return self.response_class(
+            content=tilejson_data,
+            **response_kwargs,
+        )
 
 
 class TileJsonMixin(AbstractTileJsonMixin):
-
     def get_tile_path(self):
         return reverse("geostore:layer-tiles-pattern", args=[self.object.pk])
 
     @property
     def layers(self):
-        return [self.object]
+        # keep a qs result here
+        return type(self.object).objects.filter(pk__in=[self.object.pk])
```

Comment:
Since it's only one pk, why don't you do a `.filter(pk=self.object.pk)`?
Instead of executing `WHERE id IN []` it will execute `WHERE id = `, that needs less ressources but still return a queryset.

Reply:
of course

Reply:
done